BHoM_UI/Components/Engine/GetProperty.cs

```diff
@@ -87,8 +87,13 @@ public bool SetOutputTypes(object result)
             if (OutputParams.Count < 1)
                 return true;
 
-            OutputParams[0].DataType = result.GetType();
-            CompileOutputSetters();
+            Type type = result.GetType();
+            if (OutputParams[0].DataType != type)
+            {
+                OutputParams[0].DataType = result.GetType();
```

Comment:
Could use `type` instead of `result.GetType()`

Reply:
Ha, that was the initial plan. Seems like I got distracted at the wrong time there 😋